test/e2e/pkg/testnet.go

```diff
@@ -359,6 +368,38 @@ func (t Testnet) Validate() error {
 	return nil
 }
 
+func (t Testnet) validateZones(nodes []*Node) error {
+	latencyFile := filepath.Join(t.Dir, "../../latency/aws-latencies.csv")
+	zoneMatrix, err := loadZoneLatenciesMatrix(latencyFile)
+	fileNotFoundErr := errors.Is(err, fs.ErrNotExist)
+	if !fileNotFoundErr && err != nil {
+		return err
+	}
+
+	// Get list of zone ids.
+	zones := make([]ZoneID, len(zoneMatrix))
+	for zone := range zoneMatrix {
+		zones = append(zones, zone)
+	}
+
+	// Check that the zone ids of all nodes are valid when matrix file exists.
+	for _, node := range nodes {
+		if !node.ZoneIsSet() {
+			continue
+		}
+		if fileNotFoundErr {
```

Comment:
Shall we move this condition out of the `for` loop?  https://github.com/cometbft/cometbft/pull/1560/files#diff-3a52d9f2dd1e93d9770b6c41bee2faae337bbdb1b73a41db4029b5f17e8fc91aR378 looks like the right place to me. wdyt?

Reply:
It's allowed to have no latencies file if all the nodes have no zone. But to check this condition, we need to inspect every node, and then return error if a node has a zone and there is no file.